Restore hunk from 2.6.4 that sets default for `$TMPDIR`

Fixes:

    mkdir: cannot create directory '/prepare-kernel-10482': Permission denied


--- a/scripts/prepare-kernel.sh
+++ b/scripts/prepare-kernel.sh
@@ -238,6 +238,9 @@ fi
 # Create an empty output patch file, and initialize the temporary tree.
 if test "x$output_patch" != "x"; then
 
+    if test -z "$TMPDIR"; then
+	TMPDIR=/tmp
+    fi
     temp_tree=$TMPDIR/prepare-kernel-$$
     if ! mkdir $temp_tree; then
 	echo Temporary directory $temp_tree already exists, aborting.
